firestore/client.go

```diff
@@ -127,12 +159,25 @@ func (c *Client) path() string {
 }
 
 func withResourceHeader(ctx context.Context, resource string) context.Context {
-	md, _ := metadata.FromOutgoingContext(ctx)
+	md, ok := metadata.FromOutgoingContext(ctx)
```

Comment:
can you explain what this change is and why it's needed? I would expect only the new header to be added, rather than this be modified.

Reply:
This was added to resolve : https://github.com/googleapis/google-cloud-go/pull/5331#discussion_r1263946968

Earlier if metadata did not get created in FromOutgoingContext method (https://github.com/grpc/grpc-go/blob/1d16ef5bd8fc1728072ffb2056d78c8659a5f5c5/metadata/metadata.go#L260), the md would be nil and further code would fail.

Now, I have created a new metadata object if there is none